test/consumption/2019-10-01/armconsumption/zz_generated_models.go

```diff
@@ -0,0 +1,4728 @@
+	populate(objectMap, "usedHours", r.UsedHours)
+	return json.Marshal(objectMap)
+}
+
+// UnmarshalJSON implements the json.Unmarshaller interface for type ReservationDetailProperties.
+func (r *ReservationDetailProperties) UnmarshalJSON(data []byte) error {
+	var rawMsg map[string]json.RawMessage
+	if err := json.Unmarshal(data, &rawMsg); err != nil {
+		return err
+	}
+	for key, val := range rawMsg {
+		var err error
+		switch key {
+		case "instanceFlexibilityGroup":
+			err = unpopulate(val, &r.InstanceFlexibilityGroup)
+			delete(rawMsg, key)
+		case "instanceFlexibilityRatio":
+			err = unpopulate(val, &r.InstanceFlexibilityRatio)
+			delete(rawMsg, key)
+		case "instanceId":
+			err = unpopulate(val, &r.InstanceID)
+			delete(rawMsg, key)
+		case "kind":
+			err = unpopulate(val, &r.Kind)
+			delete(rawMsg, key)
+		case "reservationId":
+			err = unpopulate(val, &r.ReservationID)
+			delete(rawMsg, key)
+		case "reservationOrderId":
+			err = unpopulate(val, &r.ReservationOrderID)
+			delete(rawMsg, key)
+		case "reservedHours":
+			err = unpopulate(val, &r.ReservedHours)
+			delete(rawMsg, key)
+		case "skuName":
+			err = unpopulate(val, &r.SKUName)
+			delete(rawMsg, key)
+		case "totalReservedQuantity":
+			err = unpopulate(val, &r.TotalReservedQuantity)
+			delete(rawMsg, key)
+		case "usageDate":
+			var aux timeRFC3339
+			err = unpopulate(val, &aux)
+			r.UsageDate = (*time.Time)(&aux)
+			delete(rawMsg, key)
+		case "usedHours":
+			err = unpopulate(val, &r.UsedHours)
+			delete(rawMsg, key)
+		}
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ReservationDetailsListResult - Result of listing reservation details.
+type ReservationDetailsListResult struct {
+	// READ-ONLY; The link (url) to the next page of results.
+	NextLink *string `json:"nextLink,omitempty" azure:"ro"`
+
+	// READ-ONLY; The list of reservation details.
+	Value []*ReservationDetail `json:"value,omitempty" azure:"ro"`
+}
+
+// MarshalJSON implements the json.Marshaller interface for type ReservationDetailsListResult.
+func (r ReservationDetailsListResult) MarshalJSON() ([]byte, error) {
+	objectMap := make(map[string]interface{})
+	populate(objectMap, "nextLink", r.NextLink)
+	populate(objectMap, "value", r.Value)
+	return json.Marshal(objectMap)
+}
+
+// ReservationRecommendationClassification provides polymorphic access to related types.
+// Call the interface's GetReservationRecommendation() method to access the common type.
+// Use a type switch to determine the concrete type.  The possible types are:
+// - *LegacyReservationRecommendation, *ModernReservationRecommendation, *ReservationRecommendation
+type ReservationRecommendationClassification interface {
+	// GetReservationRecommendation returns the ReservationRecommendation content of the underlying type.
+	GetReservationRecommendation() *ReservationRecommendation
+}
+
+// ReservationRecommendation - A reservation recommendation resource.
+type ReservationRecommendation struct {
```

Comment:
Example fix for https://github.com/Azure/autorest.go/issues/698

Handle marshallilng/unmarshalling two parents.